### PR TITLE
[opt](cancel) Cancel get result future  immediately if query is cancelled 

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ResultReceiver.java [hunk 1]
```diff
@@ -71,25 +74,43 @@ public RowBatch getNext(Status status) throws TException {
...
+                        } else {
+                            status.setStatus(new Status(TStatusCode.CANCELLED, this.cancelReason));
+                            return null;
+                        }
+                    } catch (TimeoutException e) {
+                        LOG.warn("Query {} get result timeout, get result duration {} ms",
+                                DebugUtil.printId(this.queryId), (timeoutTs - currentTs)/1000);
+                        isCancel = true;
+                        status.setStatus(Status.TIMEOUT);
+                        updateCancelReason("Fetch date timeout");
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ResultReceiver.java:101]: fetch data

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ResultReceiver.java (hunk 1 above)
Comment (zhiqiang-hhhh) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ResultReceiver.java]: done

**Review by yiguolei — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**